src/trusted-client.js

```diff
@@ -35,6 +35,26 @@ function setRequestHeader(options, header, value) {
   options.headers[header] = value;
 }
 
+const extendErrorWithResponseFields = (err, res) => {
+  if (res) {
```

Comment:
Sometimes the error object is reconstituted using errors.reviveRemoteError and I wanted to make sure that the fields are added in that case. I can probably rework it so that this is only called from there then.